src/RestApi/StoreApi/Controllers/CartOrder.php

```diff
@@ -0,0 +1,362 @@
+<?php
+/**
+ * Cart Order controller.
+ *
+ * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Controllers;
+
+defined( 'ABSPATH' ) || exit;
+
+use \WP_Error as RestError;
+use \WP_REST_Server as RestServer;
+use \WP_REST_Controller as RestController;
+use \WP_REST_Response as RestResponse;
+use \WP_REST_Request as RestRequest;
+use \WC_REST_Exception as RestException;
+use Automattic\WooCommerce\Blocks\RestApi\StoreApi\Schemas\OrderSchema;
+
+/**
+ * Cart Order API.
+ *
+ * Creates orders based on cart contents.
+ */
+class CartOrder extends RestController {
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/store';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'cart/order';
+
+	/**
+	 * Schema class instance.
+	 *
+	 * @var object
+	 */
+	protected $schema;
+
+	/**
+	 * Setup API class.
+	 */
+	public function __construct() {
+		$this->schema = new OrderSchema();
+	}
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'  => RestServer::CREATABLE,
+					'callback' => array( $this, 'create_item' ),
+					'args'     => $this->get_endpoint_args_for_item_schema( RestServer::CREATABLE ),
+				],
+				'schema' => [ $this, 'get_public_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Converts the global cart to an order object.
+	 *
+	 * @todo Since this relies on the cart global so much, why doesn't the core cart class do this?
+	 * @todo set payment method
+	 * @todo set customer note
+	 *
+	 * Based on WC_Checkout::create_order.
+	 *
+	 * @param RestRequest $request Full details about the request.
+	 * @return RestError|RestResponse
+	 */
+	public function create_item( $request ) {
```

Comment:
Let's typehint this.

Reply:
Not possible due to https://github.com/WordPress/WordPress/blob/3d623995a8d070e608f4ff297512799cf89bb2c0/wp-includes/rest-api/endpoints/class-wp-rest-controller.php#L124